internal/command/generate.go

```diff
@@ -49,10 +51,48 @@ var CmdGenerate = &Command{
 	// We should do so by moving the clone part to maybeGetLanguageRepo - because then we'll be set up
 	// with the right image etc.
 	maybeLoadStateAndConfig: loadRepoStateAndConfig,
-	execute:                 runGenerate,
+	execute:                 executeGenerate,
 }
 
-func runGenerate(state *commandState) error {
+func runGenerate(ctx context.Context) error {
+	startTime := time.Now()
+	workRoot, err := createWorkRoot(startTime)
+	if err != nil {
+		return err
+	}
+	workRoot, err = resolveLibraryWorkRoot(workRoot)
+	if err != nil {
+		return err
+	}
+	repo, err := cloneOrOpenLanguageRepo(workRoot)
```

Comment:
This and the loadRepoStateAndConfig should be conditionalized on resolveLibraryWorkRoot not returning an empty string. (Otherwise, `repo`, `config` and `state` should all be null.)

(And yes, I regret the naming of `resolveLibraryWorkRoot`...)

Reply:
filed an issue.